client/dive-common/components/Viewer.vue

```diff
@@ -343,28 +349,54 @@ export default defineComponent({
           frameRate: meta.fps,
           originalFps: meta.originalFps || null,
         });
-        imageData.value = cloneDeep(meta.imageData) as FrameImage[];
-        videoUrl.value = meta.videoUrl;
-        datasetType.value = meta.type as DatasetType;
-
-        const trackData = await loadDetections(datasetId.value, props.revision);
-        const tracks = Object.values(trackData);
-        progress.total = tracks.length;
-        for (let i = 0; i < tracks.length; i += 1) {
-          if (i % 4000 === 0) {
-          /* Every N tracks, yeild some cycles for other scheduled tasks */
-            progress.progress = i;
+        // Load non-Default Cameras if they exist:
+        const filteredMultiCamList = multiCamList.value.filter((item) => item !== 'default');
+        if (filteredMultiCamList.length === 0) {
+          imageData.value[selectedCamera.value] = cloneDeep(meta.imageData) as FrameImage[];
+          videoUrl.value[selectedCamera.value] = meta.videoUrl || null;
+          datasetType.value = meta.type as DatasetType;
+          const trackData = await loadDetections(datasetId.value);
+          const tracks = Object.values(trackData);
+          progress.total = tracks.length;
+          for (let i = 0; i < tracks.length; i += 1) {
+            if (i % 4000 === 0) {
+              /* Every N tracks, yeild some cycles for other scheduled tasks */
+              progress.progress = i;
+              // eslint-disable-next-line no-await-in-loop
+              await new Promise((resolve) => window.setTimeout(resolve, 500));
+            }
+            insertTrack(Track.fromJSON(tracks[i]), { imported: true });
+          }
+        } else {
+          for (let i = 0; i < filteredMultiCamList.length; i += 1) {
+            const camera = filteredMultiCamList[i];
             // eslint-disable-next-line no-await-in-loop
-            await new Promise((resolve) => window.setTimeout(resolve, 500));
+            const subCameraMeta = await loadMetadata(`${baseMulticamDatasetId.value}/${camera}`);
+            imageData.value[camera] = cloneDeep(subCameraMeta.imageData) as FrameImage[];
+            videoUrl.value[camera] = subCameraMeta.videoUrl || null;
+            addCamera(camera);
+            // eslint-disable-next-line no-await-in-loop
+            const camTrackData = await loadDetections(`${baseMulticamDatasetId.value}/${camera}`);
+            const camTracks = Object.values(camTrackData);
+            progress.total = camTracks.length;
+            for (let j = 0; j < camTracks.length; j += 1) {
+              if (j % 4000 === 0) {
+              /* Every N tracks, yeild some cycles for other scheduled tasks */
+                progress.progress = j;
+                // eslint-disable-next-line no-await-in-loop
+                await new Promise((resolve) => window.setTimeout(resolve, 500));
+              }
+              insertTrack(Track.fromJSON(camTracks[j]), { imported: true, cameraName: camera });
```

Comment:
We should avoid this logic duplication.

We can handle this by bumping the metadata schema version number and doing a meta.json migration when v1 is loaded so that Viewer.vue doesn't have to deal with this case.

In web, we can do a migration in mongo.

Reply:
Not important to resolve in this PR, just wanted to comment.